code/modules/events/vent_clog.dm

```diff
@@ -0,0 +1,295 @@
+	. = ..()
+	if(!.)
+		return
+	for(var/obj/machinery/atmospherics/components/unary/vent_pump/vent in GLOB.machines)
+		var/turf/vent_turf = get_turf(vent)
+		if(vent_turf && is_station_level(vent_turf.z) && !vent.welded)
+			return TRUE //make sure we have a valid vent to spawn from.
+	return FALSE
+
+/datum/round_event/vent_clog
+	announce_when = 10
+	announce_chance = 90
+	end_when = 600
+
+	///Vent selected for the event.
+	var/obj/machinery/atmospherics/components/unary/vent_pump/vent
+	///What mob will be spawned
+	var/mob/spawned_mob = /mob/living/basic/cockroach
+	///Cap on the number of spawned mobs that can be alive at once.
+	var/maximum_spawns = MOB_SPAWN_MINIMUM
+	///Interval between mob spawns.
+	var/spawn_delay = 10
+	///Used to track/limit produced mobs.
+	var/list/living_mobs = list()
+	///The list of decals we will choose from to spawn when producing a mob
+	var/list/filth_spawn_types = list()
+
+/datum/round_event/vent_clog/announce()
+	priority_announce("Minor biological obstruction detected in the ventilation network. Blockage is believed to be in the [get_area_name(vent)].", "Custodial Notification")
+
+/datum/round_event/vent_clog/setup()
+	vent = get_vent()
+	spawned_mob = get_mob()
+	end_when = rand(300, 600)
+	maximum_spawns = rand(MOB_SPAWN_MINIMUM, 10)
+	spawn_delay = rand(10, 15)
+	filth_spawn_types = list(
+		/obj/effect/decal/cleanable/vomit,
+		/obj/effect/decal/cleanable/insectguts,
+		/obj/effect/decal/cleanable/oil,
+	)
+
+/datum/round_event/vent_clog/start()
+	clog_vent()
+	announce_to_ghosts(vent)
+
+/datum/round_event/vent_clog/tick() //Checks if spawn_interval is met, then sends signal to vent to produce a mob.
+	if(activeFor % spawn_delay == 0)
+		life_check()
+		if(living_mobs.len < maximum_spawns)
+			produce_mob()
+
+/datum/round_event/vent_clog/end() //No end announcement. If you want to take the easy way out and just leave the vent welded, you must open it at your own peril.
+	vent = null
+	living_mobs.Cut()
+
+/**
+ * Selects which mob will be spawned for a given vent clog event.
+ *
+ * Creates a static list of mobs, which is different based on the severity of the event being run, and returns a pick() of it.
+ */
+
+/datum/round_event/vent_clog/proc/get_mob()
+	var/static/list/mob_list = list(
+		/mob/living/basic/butterfly,
+		/mob/living/basic/cockroach,
+		/mob/living/basic/giant_spider/maintenance,
+		/mob/living/basic/mouse,
+	)
+	return pick(mob_list)
+
+/**
+ * Finds a valid vent to spawn mobs from.
+ *
+ * Randomly selects a vent that is on-station and unwelded. If no vents are found, the event
+ * is immediately killed.
+ */
+
+/datum/round_event/vent_clog/proc/get_vent()
+	var/list/vent_list = list()
+	for(var/obj/machinery/atmospherics/components/unary/vent_pump/vent in GLOB.machines)
+		var/turf/vent_turf = get_turf(vent)
+		if(vent_turf && is_station_level(vent_turf.z) && !vent.welded)
+			vent_list += vent
+
+	if(!length(vent_list))
+		kill()
+		CRASH("Unable to find suitable vent.")
```

Comment:
shouldn't this go before the forloop, in order to be a neat early return

Reply:
the forloop populates the list with all valid scrubbers. It is here to make sure the list has scrubbers in it, so we don't pick/return from an empty list.